src/main/kotlin/com/exactpro/th2/check1/configuration/Check1Configuration.kt

```diff
@@ -27,12 +30,12 @@ class Check1Configuration(
     @field:JsonProperty(value = "cleanup-older-than", defaultValue = "60")
     val cleanupOlderThan: Long = 60L,
 
-    @field:JsonProperty(value = "max-event-batch-content-size", defaultValue = "1048576")
-    val maxEventBatchContentSize: Int = 1048576,
-
     @field:JsonProperty(value = "cleanup-time-unit", defaultValue = "SECONDS")
     val cleanupTimeUnit: ChronoUnit = ChronoUnit.SECONDS,
 
+    @field:JsonProperty(value = "cleanup-interval-millis", defaultValue = "1000")
+    val cleanupIntervalMillis: Long = 1000L,
```

Comment:
Maybe we could compute this based on the submitted tasks. We could find the earliest submitted task, take its submission time and add `cleanupTime` to that time. This will be the next time we should check for tasks to clean up.
The introduction of another parameter that should be configured by the user does not bring any benefits to us.

Reply:
This will not reduce number of cleanups significantly. We submit thousands rules a second and always have some rules for cleanup. We could use fixed interval of 1 second and remove config parameter.

Reply:
I am sorry, but I don't see any correlation between the 1k rule/s and the cleanup timeout. If all these 1k rules are submitted with unique chain IDs we cannot clean up them before the cleanup time has elapsed. Otherwise, users won't be able to use them for chaining other rules

Reply:
If we'll submit 1k rule/s we will get about 1k/s timeout expirations. So, average time to expiration of earliest submitted task after previous cleanup will be ~1ms. And I'm trying to make interval between cleanupds at least 1 sec (or `cleanup-interval-millis`).

Reply:
Okay, I believe I got you. So the concern here is that if rules are submitted with a rate of 1k rule/s eventually, all these rules will be cleaned up one by one and you want to introduce a minimal delay between two clean ups. Please, correct me here.
I think we need to rename the parameter to 'cleanup-cooldown-ms' or 'cleanup-quiet-period-ms' because 'cleanup-interval' looks more like a cleanup scheduling parameter